### PR TITLE
Add charset preference when bulk importing vBulletin

### PR DESCRIPTION
**Pull request by quangbuule**
This works for my case with`DB_CHARSET=latin1`.
The charset converting would be disabled if not set `DB_CHARSET`.

### CODE REVIEW

File: script/bulk_import/vbulletin.rb [hunk 1]
```diff
@@ -5,6 +5,44 @@
 class BulkImport::VBulletin < BulkImport::Base
 
   SUSPENDED_TILL ||= Date.new(3000, 1, 1)
+  CHARSET_MAP = {
+    "armscii8" => nil,
```
Comment (ZogStriP) [inline script/bulk_import/vbulletin.rb]: Where did you get that list from? Do we need it? I mean, can't we just pass the string to the `encode` method? It's weird this isn't supported by the standard library...

File: script/bulk_import/vbulletin.rb [hunk 2]
```diff
@@ -13,10 +51,19 @@ def initialize
     username = ENV["DB_USERNAME"] || "root"
     password = ENV["DB_PASSWORD"]
     database = ENV["DB_NAME"] || "vbulletin"
+    charset  = ENV["DB_CHARSET"] || "utf8"
```
Comment (ZogStriP) [inline script/bulk_import/vbulletin.rb]: Is there a way to automatically detect the charset used?

File: script/bulk_import/vbulletin.rb (hunk 1 above)
Comment (quangbuule) [inline script/bulk_import/vbulletin.rb]: Old version of mysql2:
http://www.rubydoc.info/gems/mysql2/0.3.11/Mysql2/Client

File: script/bulk_import/vbulletin.rb (hunk 2 above)
Comment (quangbuule) [inline script/bulk_import/vbulletin.rb]: Getting automatically is not correct in some cases. I mean, the database, even the table could be `utf8_general_ci`, but the data that the vBulletin saved still in `latin1`.

File: script/bulk_import/vbulletin.rb (hunk 1 above)
Comment (quangbuule) [inline script/bulk_import/vbulletin.rb]: Charset of mysql2 client and the ruby standard lib are not consistent. For example, mysql2 accept "utf8" while Ruby standard using "utf-8" (will throw an error if we put "utf8").
That's why there is CHARSET_MAP there.

**Comment by ZogStriP** [comment]
Thanks a lot!

**Pull request merged**